src/App.tsx

```diff
@@ -1,39 +1,70 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerClockNameId = 0;
+
+  componentDidMount(): void {
+    this.timerClockNameId = window.setInterval(this.changeClockName, 3300);
+
+    document.addEventListener('contextmenu', this.hideWatch);
+    document.addEventListener('click', this.showWatch);
+  }
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentWillUnmount(): void {
+    clearInterval(this.timerClockNameId);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.removeEventListener('contextmenu', this.hideWatch);
+    document.removeEventListener('click', this.showWatch);
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  hideWatch = (event: Event) => {
+    event.preventDefault();
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    this.setState({ hasClock: false });
```

Comment:
now, the state will change everytime when I press right-click, try to optimize your handler a bit